### PR TITLE
fix(mdDialog): if targetEvent element is hidden when dialog closes, shrink to the element's original position

### PR DESCRIPTION
**Pull request by dozingcat**

### CODE REVIEW

**Comment by dozingcat** [comment]
This is a fix for https://github.com/angular/material/issues/3541

**Comment by ThomasBurleson** [comment]
@dozingcat - can you provide a unit test case or a CodePen to validate these changes?

**Comment by dozingcat** [comment]
The codepen at http://codepen.io/anon/pen/BNYpEZ shows the current behavior. This is the local file I used to confirm the new behavior:
```html
<html>
<head>
<link rel="stylesheet" href="../material/dist/angular-material.css">
<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.1/angular.min.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.1/angular-animate.min.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.1/angular-aria.min.js"></script>
<script src="../material/dist/angular-material.js"></script>
<style>
    .menu {
      background: lightblue;  
    }
    .menu-contents {
      display: none;
      padding: 10px;
    }
    .menu:hover .menu-contents {
      display: block;
    }
    md-dialog {
      padding: 20px;
    }
</style>
<script>
  angular
    .module('demoApp', ['ngMaterial'])
    .controller("DemoController", function($scope, $mdDialog) {
        $scope.showDialog = function(e) {
          $mdDialog.show({
            template: "<md-dialog>Event: " + e + "</md-dialog>",
            clickOutsideToClose: true,
            targetEvent: e
          })
        };
    }
  );
</script>
</head>
<body>
<div ng-app="demoApp" 
     ng-controller="DemoController"
     style="float:right; padding: 20px;">
  <md-button ng-click="showDialog($event)">Open dialog</md-button>
  <p></p>
  <div class="menu">
    Menu
    <div class="menu-contents">
      <a ng-click="showDialog($event)">Open from menu</a>
    </div>
  </div>
</div>
</body>
</html>

**Comment by ThomasBurleson** [comment]
@dozingcat - that was helpful; and the demo helps me understand a usage scenario.
